lib/ssh_scan/database.rb

```diff
@@ -0,0 +1,26 @@
+# API Databases
+require 'ssh_scan/database/mongo'
+require 'ssh_scan/database/sqlite'
+
+module SSHScan
+  class DatabaseConfig
+    def initialize(opts = {})
+      @@db_type = if !opts || opts.empty?
+                   'sqlite'
+                 else
+                   opts['database']['type']
+                 end
+    end
+
+    def self.from_config_file
```

Comment:
@royalharsh I think all this file loading is getting overly confusing.

I'd like to propose we do something like this, with maybe even bringing the config loading out to the API level, since having DB configs and API configs is sort of odd behavior.  I've also mocked up an example on how we can make DB operations a little more transparent via a strong abstraction layer..

https://gist.github.com/claudijd/4004ad6c601d6b199bdf704e022857d2